src/Microsoft.ML.Transforms/Text/TextCatalog.cs

```diff
@@ -72,7 +72,8 @@ public static class TextCatalog
         /// and append another marker character, <see langword="0x03"/>, to the end of the output vector of characters.</param>
         /// <param name="columns">Pairs of columns to run the tokenization on.</param>
 
-        public static TokenizingByCharactersEstimator TokenizeIntoCharactersAsKeys(this TransformsCatalog.TextTransforms catalog,
+        [BestFriend]
+        internal static TokenizingByCharactersEstimator TokenizeIntoCharactersAsKeys(this TransformsCatalog.TextTransforms catalog,
             bool useMarkerCharacters = CharTokenizingDefaults.UseMarkerCharacters,
```

Comment:
Looks nice. Just curious. How did you find out all APIs working with ColumnOptions?

Reply:
I looked for all occurrences of `ColumnOptions[]`, `ColumnOptions[] columns`, `params ColumnOptions[]` and filtered for those with catalogs.
Than searched for all the `public.*class ColumnOptions` to find the definitions of the `ColumnOptions` classes and make them internal. 

---
In reply to: [266102562](https://github.com/dotnet/machinelearning/pull/2959#discussion_r266102562) [](ancestors = 266102562)